### ISSUE TITLE
Kolibri server crashes when importing 3 channels at once

### ISSUE THREAD

**Issue by marcellamaki** [learningequality/kolibri#9229]
## Observed behavior
From 0.15.2-a0 bug bash

## Context

 * Kolibri version 0.15.x (0.15.2-a0)
 * Operating system MacOs
 * Browser Chrome

**Comment by marcellamaki** [comment]
@jtamiace -- can you provide any more details about which channels you were importing? And was it full channels, or selected resources?

**Comment by jtamiace** [comment]
I started an import of the entire Khan Academy channel and while that was going, I started importing all of "HP Life - Courses (English)" and I think partial selection of Sikana.

**Comment by jtamiace** [comment]
Log files here
[kolibri-2022-03-25.txt](https://github.com/learningequality/kolibri/files/8366007/kolibri-2022-03-25.txt)

**Comment by rtibbles** [comment]
Main culprit appears to be:
```
IOError: [Errno 24] Too many open files: u'/Users/jessica/.kolibri/content/storage/c/7/c741068bd6fe10f6a47dc04323d3bb83.png'
```

Looks like our parallel download implementation is not properly sensitive to open file handler limits.

**Comment by rtibbles** [comment]
Flagging as a regression, as this is something that was introduced by parallel downloading work introduced in 0.14.x.

**Comment by rtibbles** [comment]
This should have been fixed by #9242
